### DISCUSSION TITLE
Game lagging if custom hitsounds turned on?

### DISCUSSION THREAD

**Discussion by NatureHog** [ppy/osu#17210] [Q&A] [answered]
I've got a really strange problem.

On specific maps, which have multiple custom hitsounds the game stutters for a moment, right when the hitsounds are changing from one to another. This makes some beatmaps (almost) impossible to beat with "Beatmap Hitsounds" turned on...

I came across two examples of this happening in my current database:
[First Beatmap](https://osu.ppy.sh/beatmapsets/942641#osu/2056701)
[Second Beatmap](https://osu.ppy.sh/beatmapsets/1309729#osu/2716381)

Here are the logs of playing the first one above first with custom hitsounds then without, even though I personally didn't find anything helpful:
[performance.log](https://github.com/ppy/osu/files/8227646/performance.log)
[runtime.log](https://github.com/ppy/osu/files/8227647/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8227648/updater.log)
[database.log](https://github.com/ppy/osu/files/8227649/database.log)
[network.log](https://github.com/ppy/osu/files/8227650/network.log)

I am using Windows 11 with the newest .NET installed, if that is of any help.

**Comment by bdach** [comment]
I've checked and am not seeing anything of the sort here. Can you attach a video of this, preferably with the frame graph showing (press ctrl-f11 twice)?

**Reply by NatureHog** [reply]
I went through the exact same steps as before, including the same beatmaps with the same difficulty.
Here is the recording:

https://user-images.githubusercontent.com/55314658/157837804-7e3ae2a8-8749-456b-8615-5e65277ebfca.mp4

Here are the logs to go alongside this recording:
[Logs.zip](https://github.com/ppy/osu/files/8230058/Logs.zip)

**Reply by frenzibyte** [reply]
It seems the recording you attached is [corrupted](https://cln.sh/y2DZH9).

**Reply by NatureHog** [reply]
I'm sorry, I had some problems uploading it to Github...
I hope a OneDrive link to it will suffice:
[Recording.mp4](https://1drv.ms/v/s!AqwZ15G5dYM5gr5vIELkxuVtCuG2Tg?e=2ZRAIi)

**Reply by NatureHog** [reply]
I just tried something different: I wanted to find out if the hitsounds weren't loading in RAM (or something similar, I don't know the techincal details...) and recorded it.
My thought process was basically:
1. Reboot PC
2. Play the same levels as above twice in a row
3. If the first time is laggier compared to the second time, then the hitsounds wouldn't be loaded fast enough the first time, bur the second time the hitsounds are already in RAM, so it loads them faster

I don't know C# at all, so I don't know if it's actually saved in RAM in practice, but my basic theory behind it was correct: The first time was much laggier than the second time.

Recording with frame graph here (via OneDrive again):
[RecordingAfterReboot.mp4](https://1drv.ms/v/s!AqwZ15G5dYM5gr5w6dlkJH9S3mIRPw?e=fHr1Md)
Logs here:
[LogsAfterReboot.zip](https://github.com/ppy/osu/files/8237875/LogsAfterReboot.zip)

**Reply by peppy** [reply]
I can reproduce this issue (already have in the past and been meaning to look into it). Will take this.

**Comment by frenzibyte** [comment, marked as answer]
Escalated to #17241

**Comment by peppy** [comment]
This should be resolved now @NatureHog can you please test again?

**Reply by NatureHog** [reply]
Yes, in my testing it seems resolved.
